explorer/src/lib/dusk/string/decodeHexString.js

```diff
@@ -0,0 +1,31 @@
+import { hexToBytes, isValidHex } from "./";
+
+/**
+ * Decodes a hexadecimal string to utf-8 and formats it if it's a JSON string
+ *
+ * @param {string} value
+ * @returns {string}
+ */
+function decodeHexString(value) {
+  if (!isValidHex(value)) {
+    return value;
+  }
+
+  const bytes = hexToBytes(value);
+  const decoder = new TextDecoder();
+  const decodedString = decoder.decode(bytes);
+
+  //Check whether the string contains almost all visible Unicode characters up to the end of the BMP
+  /* eslint-disable-next-line no-control-regex */
+  if (/^[\u0000-\u007F\u00A0-\uFFFF]*$/.test(decodedString)) {
```

Comment:
Perhaps the regex here is too broad – it allows control and invisible characters. Perhaps we could be more restrictive and go with something like `/^[\u0020-\u007E\u00A0-\u00FF]*$/` instead.